x-pack/plugins/security_solution/common/detection_engine/prebuilt_rules/model/diff/three_way_diff/three_way_diff_outcome.ts

```diff
@@ -29,14 +30,25 @@ export enum ThreeWayDiffOutcome {
 }
 
 export const determineDiffOutcome = <TValue>(
-  baseVersion: TValue,
+  baseVersion: TValue | MissingVersion,
   currentVersion: TValue,
   targetVersion: TValue
 ): ThreeWayDiffOutcome => {
   const baseEqlCurrent = isEqual(baseVersion, currentVersion);
   const baseEqlTarget = isEqual(baseVersion, targetVersion);
   const currentEqlTarget = isEqual(currentVersion, targetVersion);
 
+  if (baseVersion === MissingVersion) {
+    /**
+     * We couldn't find the base version of the rule in the package so further
+     * version comparison is not possible. We assume that the rule is not
+     * customized and the value can be updated if there's an update.
+     */
+    return currentEqlTarget
+      ? ThreeWayDiffOutcome.StockValueNoUpdate
+      : ThreeWayDiffOutcome.StockValueCanUpdate;
+  }
```

Comment:
WRT

> The above logic has considerable implications. When a historical version of a rule is unavailable, we assume that the rule had no modifications. Therefore, in cases where it has been modified, the upgrade review endpoint would still return "no conflicts detected", and users risk losing all their edits.

In these instances, I'm curious if we should have some UI/UX to help mitigate the issue in the event it actually happens. Whether that's encouraging the user to do a quick export of existing rules, a system-triggered temporary auto-backup of the rules in question, or something along those lines? Depending on how the storage of historical rules goes, perhaps this is uncommon enough to not even be necessary?

Reply:
Given that users could lose all their modifications without the ability to restore them, I think it makes sense to implement some safeguards on the client side. It's not a big concern, though, as long as we do not allow them to customize prebuilt rules. So, maybe we should wait and see how the adoption of historical rules goes. If we see a substantial number of packages without historical versions or not all versions available for every rule, then we'll need to implement some workarounds in the UI.